### ISSUE TITLE
Cannot specify charset on 'Content-Type' header

### ISSUE THREAD

**Issue by AlexandreRoba** [zino-app/graphql-flutter#94]
**To Reproduce**
Querying a graphql endpoints that returns accents such as "Belvédère". Instead we get a string with "BelvÃ©dÃ¨re"

**Expected behavior**
Find the string "belvédère" in the jsonResponse Map after decoding (line 76 client.dart).
```dart
final Map<String, dynamic> jsonResponse = json.decode(response.body);
```

The header parameters are hardcoded and the 'Content-Type' is set to application/json.
In order to support the accents we should be able to specify the charset such as:
'Content-Type: application/json; charset=utf-8'

According to dart documentation:
```dart
/// The body of the response as a string. This is converted from [bodyBytes]
  /// using the `charset` parameter of the `Content-Type` header field, if
  /// available. If it's unavailable or if the encoding name is unknown,
  /// [latin1] is used by default, as per [RFC 2616][].
  ///
  /// [RFC 2616]: http://www.w3.org/Protocols/rfc2616/rfc2616-sec3.html
  String get body => _encodingForHeaders(headers).decode(bodyBytes);
```

**Comment by HofmannZ** [comment]
@AlexandreRoba Thanks for reporting the issue. Will see if I can fix it today.

**Comment by HofmannZ** [comment]
@AlexandreRoba 
> The header parameters are hardcoded and the 'Content-Type' is set to application/json.

Which version are you using?

In `1.0.0` we allow you to extend the headers your self:

```dart
HttpLink link = HttpLink(
  uri: 'https://api.github.com/graphql',
  headers: <String, String>{
    'Authorization': 'Bearer <YOUR_PERSONAL_ACCESS_TOKEN>',
    'Content-Type': 'application/json; charset=utf-8',
   },
);
```

Also reading trough the docs for the `http` package we use in the `HttpLink`. There they say the following about the `body` method on the the `Response` class.: 

> The body of the response as a string. This is converted from bodyBytes using the charset parameter of the Content-Type header field, if available. If it's unavailable or if the encoding name is unknown, latin1 is used by default, as per RFC 2616. [...]

So the `Response.body` should already be converted to the correct `charset` before we [parse the the JSON to a map](https://github.com/zino-app/graphql-flutter/blob/9110fb15354ea487664cab5a2597c20a98a8f5f2/lib/src/link/http/link_http.dart#L198).

**Comment by AlexandreRoba** [comment]
Hi @HofmannZ thanks for the quick response.

I'm currently using version 0.9.3.

When is the 1.0.0 expected to be releases?

Your are absolutely right about the decoding of the body. If the charset is specified it will be used to decode the body according to the documentation. This is why I pasted the last bit of code in my previous post :) 

This seems to be possible with HttpLink. I will have a look at it once it is released.

Thanks for your help.

**Comment by HofmannZ** [comment]
@AlexandreRoba You can already try [1.0.0](https://github.com/zino-app/graphql-flutter/tree/next) out out today. See the [upgrade guide](https://github.com/zino-app/graphql-flutter/tree/next#upgrading-from-0xx) to for breaking changes.

**Comment by HofmannZ** [comment]
Fixed with 1.0.0.
